pages/compare.py

```diff
@@ -114,7 +114,16 @@ def list_models(stub: V2Stub,
 input_auth._pat = pat
 input_stub = create_stub(input_auth)
 
-auth = ClarifaiAuthHelper.from_streamlit(st)
+module_query_params = st.experimental_get_query_params()
+# Check if the user is logged in. If not, use internal PAT.
+if module_query_params.get("pat", "") == "" and module_query_params.get("token", "") == "":
```

Comment:
does the frontend actually send empty pats in this case? i thought they were sending a PAT. I think we should change this to just ignore the PAT if there is a PAT already provided. 

I plan to make a change to module manager to let users set secrets which would work in that way. The order would be something like: 
- env variables take precendence
- then the config file on disk
- then query params

So that way even if a query param is passed it's ignored if secrets are hard coded. If you want to make this module work that way that would be great. i can clean it up later to use whatever config format we come up with for disk. 

Then you can make input_stub and stub the same so this module is a little simpler and just uses one PAT.

Reply:
discussed on slack, this is better for now, frontend will be updated to send no PAT when not logged in.